Assets/Scripts/RammingSpecialCreature.cs

```diff
@@ -13,11 +13,55 @@ public class RammingSpecialCreature : MonoBehaviour
     public int speed = 3;
     public Transform target;
 
+    public bool useMouseClick;
+
+    // For clicking
+    public GameObject player;
+    public bool _isPainted;
+
+    public string paintColour1;
+    public string paintColour2;
+    public int paintQuantity1;
+    public int paintQuantity2;
+
+    private UpdateUI _updateUI;
+    private bool _isMouseClicked;
+
+    private bool _isMouseOver;
+
     void Start()
     {
         _levelManager = FindObjectOfType<LevelManager>();
         _material = GetComponentInChildren<Renderer>().material;
         _originalColour = _material.color;
+        _updateUI = FindObjectOfType<UpdateUI>();
+    }
+
+    void Update()
```

Comment:
This method has way too many nested ifs. I suggest combining them and storing the big condition checks in variables again for readability

Reply:
Thanks for the suggestion updated. Though I probably made a worse code smell by putting it in one method :') feel free to change